atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManagerImpl.java

```diff
@@ -114,6 +115,34 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         return numTimesFailed > NUM_RETRIES;
     }
 
+    public <C extends PreCommitCondition, E extends Exception> ServiceWriteTransaction getWriteTransaction(C condition)
+            throws E {
+        checkOpen();
+        try {
+            RawTransaction tx = setupRunTaskWithConditionThrowOnConflict(condition);
+            return new ServiceWriteTransaction(tx.delegate()) {
+                @Override
+                public void commit() {
+                    tx.commit();
+                    finishRunTaskWithLockThrowOnConflict(tx, transaction -> null);
+                }
+            };
+        } finally {
+            condition.cleanup();
+        }
+    }
+
+    public <C extends PreCommitCondition, E extends Exception> ServiceReadOnlyTransaction getReadTransaction(
+            C condition) throws E {
+        checkOpen();
+        try {
+            SnapshotTransaction tx = createReadOnlySnapshotTransaction(condition);
```

Comment:
1. We throw for THOROUGH tables for readOnly tasks on `ReadTransaction`. Since we create a `SnapshotTransaction`, we this check.

2. As requested we don't grab the immutableTs lock.

Reply:
I guess the contract here ends up being weird... We skip the THOROUGH checks that provide some safety by assuming that you're holding the immutableTs lock, but leave that up to the client. We'll need to be very explicit in docs that anyone using this with THOROUGH sweep must both acquire the lock themselves and have a precommit condition that verifies that the lock is still held.

Reply:
Actually, I think we should just consider removing the immutableTs lock checks on every read to THOROUGH sweep tables. It seems like a reasonable contract that no one should be making use of state from a failed transaction. To even do so, you have to try/catch around the transaction and store the read data in final vars.

Reply:
Will do that in a follow-up PR